### ISSUE TITLE
S = "${WORKDIR}/git" error

### ISSUE THREAD

**Issue by thomas-roos** [priv-kweihmann/oelint-adv#463]
Hi,
in version: oelint-adv 3.26.5
I get this error: error:oelint.var.override:Variable 'S' is set by XXX
Means that this construct 

S = "${WORKDIR}/git" 

is wrong, which seem to be used in many recipes - what is the suggested way of doing this? Or is this a false positive / error?

minimal example
```
oelint-adv poky/meta/recipes-devtools/file/file_5.41.bb  2>&1 >/dev/null | grep Variable | grep error
/home/ubuntu/data/yocto/yocto_kirkstone/poky/meta/recipes-devtools/file/file_5.41.bb:1:error:oelint.var.override:Variable 'S' is set by 'file_5.41.bb'
```

Cheers, Thomas

**Comment by priv-kweihmann** [comment]
```
Parsing file_5.41.bb
.../file_5.41.bb:1:info:oelint.var.suggestedvar.BUGTRACKER:Variable 'BUGTRACKER' should be set
.../file_5.41.bb:1:info:oelint.var.suggestedvar.AUTHOR:Variable 'AUTHOR' should be set
.../file_5.41.bb:1:info:oelint.var.suggestedvar.CVE_PRODUCT:Variable 'CVE_PRODUCT' should be set
.../file_5.41.bb:2:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
.../file_5.41.bb:3:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
.../file_5.41.bb:14:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
.../file_5.41.bb:15:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 11 set, 4 desirable
.../file_5.41.bb:38:warning:oelint.tabs.notabs:Don't use tabs use spaces
.../file_5.41.bb:42:warning:oelint.tabs.notabs:Don't use tabs use spaces
.../file_5.41.bb:47:warning:oelint.tabs.notabs:Don't use tabs use spaces
.../file_5.41.bb:53:warning:oelint.var.order.PROVIDES:'PROVIDES' should be placed before 'BBCLASSEXTEND'
.../file_5.41.bb:56:error:oelint.vars.inconspaces:Assignment should be 'VAR:append = " foo"' not 'VAR:append = "foo"'
```

This is what I get with the latest (3.26.5) version for this recipe.
So I'm not sure where that should come from.

**Comment by thomas-roos** [comment]
```
git clone git://git.yoctoproject.org/poky -b  kirkstone
cd poky && git checkout f754b5d45b504e7486364ea9cfcb5f96086084bc && cd ..
oelint-adv poky/meta/recipes-devtools/file/file_5.41.bb

...
Parsing poky/meta/recipes-devtools/file/file_5.41.bb                                                                                                                                                                                                                                      
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.CVE_PRODUCT:Variable 'CVE_PRODUCT' should be set                                                                                                                           
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:1:error:oelint.var.override:Variable 'S' is set by 'file_5.41.bb'                                                                                                                                         
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.AUTHOR:Variable 'AUTHOR' should be set                                                                                                                                     
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.BUGTRACKER:Variable 'BUGTRACKER' should be set                                                                                                                             
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:2:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable                                                                                                 
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:3:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable                                                                                                 
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:14:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable                                                                                                
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:15:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 11 set, 4 desirable                                                                                               
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:38:warning:oelint.tabs.notabs:Don't use tabs use spaces                                                                                                                                                   
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:42:warning:oelint.tabs.notabs:Don't use tabs use spaces                                                                                                                                                   
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:47:warning:oelint.tabs.notabs:Don't use tabs use spaces                                                                                                                                                   
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:53:warning:oelint.var.order.PROVIDES:'PROVIDES' should be placed before 'BBCLASSEXTEND'                                                                                                                   
/home/ubuntu/data/yocto/yocto_test/poky/meta/recipes-devtools/file/file_5.41.bb:56:error:oelint.vars.inconspaces:Assignment should be 'VAR:append = " foo"' not 'VAR:append = "foo"'                                                                                                      
ubuntu@ip-172-31-25-148:~/data/yocto/yocto_test$ oelint-adv --version                                                                                                                                                                                                                     
oelint-adv 3.26.5                                                                                                                                                                                                                                                                         
ubuntu@ip-172-31-25-148:~/data/yocto/yocto_test$ cd poky/                                                                                                                                                                                                                                 
ubuntu@ip-172-31-25-148:~/data/yocto/yocto_test/poky$ git log                                                                                                                                                                                                                             
commit f754b5d45b504e7486364ea9cfcb5f96086084bc (HEAD, origin/kirkstone, kirkstone)                                                                                                                                                                                                       
Author: Richard Purdie <richard.purdie@linuxfoundation.org>                                                                                                                                                                                                                               
Date:   Tue Dec 19 14:29:32 2023 +0000                                               
```

different machine - with old oelint-adv version
```
Parsing poky/meta/recipes-devtools/file/file_5.41.bb
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.CVE_PRODUCT:Variable 'CVE_PRODUCT' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.AUTHOR:Variable 'AUTHOR' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.BUGTRACKER:Variable 'BUGTRACKER' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:2:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:3:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:14:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:15:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 11 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:38:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:42:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:47:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:53:warning:oelint.var.order.PROVIDES:'PROVIDES' should be placed before 'BBCLASSEXTEND'
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:56:error:oelint.vars.inconspaces:Assignment should be 'VAR:append = " foo"' not 'VAR:append = "foo"'
throos@u182adcbddbfb55:~/Desktop$ oelint-adv --version
oelint-adv 3.22.2
```
after upgrade - same issue: 
```
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.BUGTRACKER:Variable 'BUGTRACKER' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.CVE_PRODUCT:Variable 'CVE_PRODUCT' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:info:oelint.var.suggestedvar.AUTHOR:Variable 'AUTHOR' should be set
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:1:error:oelint.var.override:Variable 'S' is set by 'file_5.41.bb'
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:2:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:3:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:14:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 0 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:15:info:oelint.vars.multilineident:On a multiline assignment, line indent is desirable. 11 set, 4 desirable
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:38:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:42:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:47:warning:oelint.tabs.notabs:Don't use tabs use spaces
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:53:warning:oelint.var.order.PROVIDES:'PROVIDES' should be placed before 'BBCLASSEXTEND'
/home/ANT.AMAZON.COM/throos/Desktop/poky/meta/recipes-devtools/file/file_5.41.bb:56:error:oelint.vars.inconspaces:Assignment should be 'VAR:append = " foo"' not 'VAR:append = "foo"'
throos@u182adcbddbfb55:~/Desktop$ oelint-adv --version
oelint-adv 3.26.5
```

any ideas what to check to support you?

**Comment by priv-kweihmann** [comment]
could you try the following please

```shell
python3 -m venv .env
. .env/bin/activate
pip3 install oelint-adv
oelint-adv poky/meta/recipes-devtools/file/file_5.41.bb
```

it might be some outdated dependency or something - I don't see what else could be potentially causing this behavior:

There was a change in https://github.com/priv-kweihmann/oelint-adv/releases/tag/3.25.0 - maybe try to test with version 3.24.0 and then 3.25.0 to see if that change introduced the behavior.
I'm still having a hard time reproducing this on my end

**Comment by thomas-roos** [comment]
In the venv with new just released version 3.26.6 it just works - on my to ubuntu 20.04 machines without venv still not.
Will try to narrow this issue down. But not today.

**Comment by thomas-roos** [comment]
But what I forgot to mention is that it failed here originally: https://github.com/aws4embeddedlinux/meta-aws/actions/workflows/oelint-adv.yml
So really wondering what is wrong. Will check and report after winter vacation...

**Comment by priv-kweihmann** [comment]
> But what I forgot to mention is that it failed here originally: https://github.com/aws4embeddedlinux/meta-aws/blob/master/.github%2Fworkflows%2Foelint-adv.yml So really wondering what is wrong. Will check and report after winter vacation...

I don't see much wrong here - but maybe you can point me to an action that failed, showing this particular error.
I got a hunch, that this could be linked to a "grouping of input files gone bad", still I'm confused why the tool behaves like that - only if one would pass the same file twice as part of the argument list, I would show this kind of error.

**Comment by thomas-roos** [comment]
This version fixed my issue without the need for venv (tested on two machines)
Might you have some idea what happend?
You can close this bug.
```
~/Desktop$ pip3 install oelint-adv --upgrade
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: oelint-adv in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (3.26.6)
Collecting oelint-adv
  Downloading oelint_adv-3.26.12-py2.py3-none-any.whl.metadata (20 kB)
Requirement already satisfied: anytree~=2.12 in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (from oelint-adv) (2.12.1)
Requirement already satisfied: colorama~=0.4 in /usr/local/lib/python3.8/dist-packages (from oelint-adv) (0.4.6)
Requirement already satisfied: oelint-parser~=2.13 in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (from oelint-adv) (2.13.0)
Requirement already satisfied: urllib3~=2.1 in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (from oelint-adv) (2.1.0)
Requirement already satisfied: wheel~=0.42 in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (from oelint-adv) (0.42.0)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from anytree~=2.12->oelint-adv) (1.14.0)
Requirement already satisfied: regex==2023.12.25 in /home/ANT.AMAZON.COM/throos/.local/lib/python3.8/site-packages (from oelint-parser~=2.13->oelint-adv) (2023.12.25)
Downloading oelint_adv-3.26.12-py2.py3-none-any.whl (86 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 86.6/86.6 kB 2.5 MB/s eta 0:00:00
Installing collected packages: oelint-adv
  Attempting uninstall: oelint-adv
    Found existing installation: oelint-adv 3.26.6
    Uninstalling oelint-adv-3.26.6:
      Successfully uninstalled oelint-adv-3.26.6
Successfully installed oelint-adv-3.26.12
~/Desktop$ oelint-adv --version   
oelint-adv 3.26.12

```

**Comment by priv-kweihmann** [comment]
Absolutely not, but happy that the issue (which i still couldn't reproduced :smile: ) auto-magically resolved itself
